### PR TITLE
[#122] Replace cookie with local storage for token

### LINKED ISSUE #122: Cannot set cookie in Hugging Face Space due to third-party cookie restrictions
We are currently unable to set cookies in our Hugging Face Space application due to third-party cookie restrictions implemented by modern web browsers. This is causing issues with our rate limt functionality.

### PR DESCRIPTION
**Pull request by kangsuhyun-yanolja**
This change addresses the issue of not being able to set cookies in our Hugging Face Space due to third-party cookie restrictions. It implements a solution using localStorage to manage tokens.

Resolves #122

### CODE REVIEW

File: rate_limit.py [hunk 1]
```diff
@@ -91,12 +91,10 @@ def set_token_server():
 
   set_token_client = """
   function(newToken) {
-    const expiresDateString = new Date(Date.now() + 24 * 60 * 60 * 1000).toUTCString();
-    document.cookie = `arena_token=${newToken}; expires=${expiresDateString};`;
+    localStorage.setItem("arena_token", newToken);
```
Comment (seungduk-yanolja) [inline rate_limit.py:94]: where is the code getting this token then?

**Comment by rifqiyan** [comment]
May I know how the token is retrieved from local storage and sent to the server?
I saw code about saving it to local storage, but don't see any change about reading it.

File: rate_limit.py (hunk 1 above)
Comment (kangsuhyun-yanolja) [reply rate_limit.py:94]: token will be stored in [token](https://github.com/yanolja/arena/pull/123/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R83-R84) param.
Each request will send this token [here](https://github.com/yanolja/arena/pull/123/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R192)

**Comment by kangsuhyun-yanolja** [comment]
@rifqiyan token will be stored in [token](https://github.com/yanolja/arena/pull/123/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R83-R84). Each request will send this token [like this](https://github.com/yanolja/arena/pull/123/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R192)

File: rate_limit.py (hunk 1 above)
Comment (seungduk-yanolja) [reply rate_limit.py:94]: The token is stored in the memory. Doesn't it mean that you don't have to save it somewhere?

Comment (kangsuhyun-yanolja) [reply rate_limit.py:94]: Then how could we send the token when a user requests..?

File: app.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def vote(vote_button, response_a, response_b, model_a_name, model_b_name,
 """
 
 with gr.Blocks(title="Yanolja Arena", css=css) as app:
-  set_token(app)
+  token = gr.Textbox(visible=False)
```
Comment (geon-yanolja) [inline app.py:83]: I have the same questions as others.
1. If it's saved here in this variable which live in memory, and this token is used through the whole app, what's the reason why you save it in local storage?
2. What's the reason why you save it in local storage when you don't re-use it? like, the initialization of token should be.. `get token from local storage -> set it as a token -> use it..`? isn't it?

Comment (kangsuhyun-yanolja) [reply app.py:83]: you're right. I'll update the code thank you!

Comment (geon-yanolja) [inline app.py:83]: what's the reason why token is a text box? like, not just string?
https://github.com/yanolja/arena/pull/123/files#diff-b2d7b04ca1a7dab749196b5980281f02920d9a1948533731f5b1a5cbd7ac5999R102
it's a plain string again so I'm confused a bit.

Comment (kangsuhyun-yanolja) [reply app.py:83]: string variables cannot be used as Gradio inputs and outputs, so I used Textbox for storing a string value.
gr.State could be an option but currently there is a bug with JS

**Review by rifqiyan — APPROVED** [review]

**Review by seungduk-yanolja — APPROVED** [review]

**Pull request merged**